src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const sortedItemsGroup: GroupsMap<T> = {};
+
+  items.reduce((acc: GroupsMap<T>, el: T) => {
+    if (!(el[key] in acc)) {
```

Comment:
So that there is no mistake, use a more understandable check:
```suggestion
    const objectKey = el[key];
    if (!sortedItemsGroup[objectKey]) { // sortedItemsGroup should be used provided that you will not use reduce
```